### PR TITLE
Use otp-ui map components

### PR DESCRIPTION
**Pull request by evansiroky**
This PR swaps out most of the map components to use the otp-ui map components and makes things compatible for the components that didn't have a replacement.

This PR depends on https://github.com/opentripplanner/otp-ui/pull/89.

### CODE REVIEW

**Comment by evansiroky** [comment]
Assigning back to myself to fix Travis problems.

**Comment by evansiroky** [comment]
I added a fix to https://github.com/opentripplanner/otp-ui/pull/89 that should resolve the Travis build errors.

**Comment by evansiroky** [comment]
Ready for review

File: lib/components/map/bounds-updating-overlay.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  componentDidMount () {
+    this.updateBounds(null, this.props)
+  }
+
+  componentDidUpdate (prevProps) {
+    this.updateBounds(prevProps, this.props)
+  }
+
+  /* eslint-disable-next-line complexity */
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js]: This method is long indeed!

File: lib/components/map/default-map.js [hunk 2]
```diff
@@ -1,83 +1,152 @@
...
+            }
+          </div>
+          <div>
+            Plan a trip:
+            <FromToLocationPicker
+              location={mapPopupLocation}
+              setLocation={this.onSetLocationFromPopup}
+            />
+          </div>
+        </div>
```
Comment (binh-dam-ibigroup) [inline lib/components/map/default-map.js]: Should this be its own component/function?

File: lib/components/map/default-map.js [hunk 3]
```diff
@@ -1,83 +1,152 @@
...
     } = this.props
+
+    const center = mapConfig && mapConfig.initLat && mapConfig.initLon
+      ? [mapConfig.initLat, mapConfig.initLon]
+      : null
+
+    const popup = mapPopupLocation && {
+      contents: (
+        <div style={{ width: 240 }}>
+          <div style={{ fontSize: 14, marginBottom: 6 }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/map/default-map.js]: With our use of styled-components, please move the styles to a separate variable.

File: lib/components/map/default-map.js [hunk 4]
```diff
@@ -1,83 +1,152 @@
...
+        <BaseMap
+          center={center}
+          maxZoom={mapConfig.maxZoom}
+          onClick={this.onMapClick}
+          popup={popup}
+          onPopupClosed={this.onPopupClosed}
+          zoom={mapConfig.initZoom || 13}
+        >
+          {/* The default overlays */}
+          <BoundsUpdatingOverlay />
```
Comment (binh-dam-ibigroup) [inline lib/components/map/default-map.js]: This is pretty cool.

File: lib/components/mobile/results-screen.js [hunk 5]
```diff
@@ -99,10 +104,10 @@ class MobileResultsScreen extends Component {
         <Row className='locations-summary' style={{ padding: '4px 8px' }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/results-screen.js]: I would move the inline styles out.

File: lib/components/map/bounds-updating-overlay.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import isEqual from 'lodash.isequal'
+// import {
+//   getItineraryBounds,
+//   getLegBounds
+// } from '@opentripplanner/core-utils/lib/itinerary'
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js]: Remove for now.

File: lib/components/map/bounds-updating-overlay.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+import { connect } from 'react-redux'
+import { MapLayer, withLeaflet } from 'react-leaflet'
+
+import {
+  getItineraryBounds,
+  getLegBounds
+} from '../../util/itinerary'
+import { getActiveItinerary, getActiveSearch } from '../../util/state'
+
+class BoundsUpdatingOverlay extends MapLayer {
```
Comment (landonreed) [inline lib/components/map/bounds-updating-overlay.js]: Component name is pretty clear, but I think we should still have javadoc here.

File: lib/components/map/connected-park-and-ride-overlay.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import { connect } from 'react-redux'
+
+import { setLocation } from '../../actions/map'
+import { parkAndRideQuery } from '../../actions/api'
+
+class ConnectedParkAndRideOverlay extends Component {
+  componentDidMount () {
+    const params = {}
+    if (this.props.maxTransitDistance) {
+      params['maxTransitDistance'] = this.props.maxTransitDistance
```
Comment (landonreed) [inline lib/components/map/connected-park-and-ride-overlay.js]: I'm confused as to why you're referencing the key in this way. Why not use dot notation?

File: lib/components/map/distance-measure.js [hunk 9]
```diff
@@ -1,21 +0,0 @@
...
-    })
-  }
-
-  componentDidMount () {
-    const { map } = this.props.leaflet
-    this.leafletElement.addTo(map)
-  }
-}
-
-export default withLeaflet(DistanceMeasure)
```
Comment (landonreed) [inline lib/components/map/distance-measure.js]: Does TriMet no longer need this?

Comment (evansiroky) [reply lib/components/map/distance-measure.js:21]: It wasn't imported anywhere, so I guess they don't use it anymore.

**Comment by evansiroky** [comment]
Ready for review again.

**Pull request merged**